zygarde.js

```diff
@@ -200,42 +235,73 @@ client.on("disconnect", evt => {
     process.exit(1);
   }
 });
+
 client.on("error", evt => console.error(evt));
 client.on("warn", info => console.warn(info));
 //client.on('debug', info => console.debug(info));
 
+// start listening to Discord
 client.on("message", async msg => {
+  // ignore messages that aren't from real users
+  // on real server (i.e. not PMs)
   if (msg.author.bot || !msg.guild) {
     return;
   }
+
   const sender = msg.member ? msg.member.displayName : msg.author.username;
+
+  // figure out where to bridge the message to
   const matching = [];
   for (const {
     zephyrClass,
     discordServer,
     connectionDirection = "",
     zephyrRelatedClasses = {}
   } of settings.classes) {
-    if (connectionDirection != ">" && discordServer == msg.guild.name) {
+    if (
+      // don't bridge if we're not going that direction
+      connectionDirection != Z2D_ONLY && 
+      // also, make sure we're bridging the right server
+      discordServer == msg.guild.name
+    ) {
+      // use a regexp to match messages of the form
+      // [-c class] [-i instance] message, which we pass onto zephyr
+      // capture groups:
+      // 1 - whole class prefix
+      // 2 - class prefix content
+      // 3 - class name
+      // 4 - whole instance prefix
+      // 5 - instance name
+      // 6 - message
```

Comment:
Thank you for commenting this. I might tweak it in another commit, only because I just realized I could use array destructuring to give names to all of these blasted indicies :P